### PR TITLE
feat: dropdown with side menu

### LINKED ISSUE #1776: Notebooks: Create Dropdown Menu Experience
We would like to update our Notebooks menu experience to utilize a dropdown menu. This update would create a dropdown experience with relevant options on a per panel basis and will require decoupling existing menu options and adding them into the drop down menu (ex: insert functions, visualization options, etc). 

In addition, we would like to change our menu icon from a gear to three dots, to better represent the user interaction. 

Eventually, we will have icons for each item in a panel's dropdown menu. For now, we can go with simple text options and we will follow with icons in a separate issue. 

## Dropdown Menu

Figma: https://www.figma.com/file/8S8IH8mimYIwanZ6keIQ6U/Notebooks-Prototype?node-id=3071%3A1738

## Acceptance Criteria

- Notebook menu is now a dropdown and includes all relevant options based on panel type.
- The drop down menu is toggled via a three dot icon.

### PR DESCRIPTION
**Pull request by ShmuelLotman**
Closes #1776

Converts the sidebar to a dropdown, but leaves the sidebar for submenus. Also fixes the padding for the scrollbar on the sidemenu so its not behind the fields
Screen Shot 2021-07-02 at 9 15 39 AM

Screen Shot 2021-07-02 at 9 15 30 AM

### CODE REVIEW

**Comment by ui-ci-bot-influx** [comment]
Metric | Score | Level
------------ | ------------- | ------------- 
performance | 57 | :red_circle:
accessibility | 96 | :green_circle:
seo | 90 | :green_circle:
pwa | null | :hankey:
best-practices | 100 | :green_circle:


[Compare your scores with master branch](https://influxdb.aws.influxdata.io/orgs/844910ece80be8bc/dashboards/073c4b95f1089000?lower=now%28%29%20-%202d&vars%5Blighthouse_score_branch%5D=master)

File: src/flows/components/Sidebar.tsx [hunk 1]
```diff
@@ -33,15 +33,29 @@ const Sidebar: FC = () => {
...
+
+const Sidebar: FC = () => {
+  const {flow, add} = useContext(FlowContext)
+  const {getPanelQueries} = useContext(FlowQueryContext)
+  const {id, show, hide, menu, showSub} = useContext(SidebarContext)
 
   const sections = ([
     {
-      title: 'Panel',
+      title: '',
```
Comment (Asalem1) [inline src/flows/components/Sidebar.tsx:58]: why no title?

File: src/flows/components/Sidebar.tsx [hunk 2]
```diff
@@ -191,24 +205,27 @@ const Sidebar: FC = () => {
...
-      <List>{sections}</List>
-    </div>
+    <DropdownMenu
+      style={{
+        width: '200px',
+        position: 'absolute',
+        top: '40px',
+        zIndex: 2,
+        padding: '10px 0px',
+      }}
```
Comment (Asalem1) [inline src/flows/components/Sidebar.tsx:225]: can this all be a class with cf variables? Why the absolute positioning? Is it absolute relative to a panel?

File: src/flows/pipes/Visualization/style.scss [hunk 3]
```diff
@@ -21,6 +21,7 @@
 
 .flow-sidebar--submenu .view-options {
   padding: 0;
+  padding-right: 15px;
```
Comment (Asalem1) [inline src/flows/pipes/Visualization/style.scss:24]: same about variable?

File: src/flows/components/panel/FlowPanel.tsx [hunk 4]
```diff
@@ -144,17 +150,18 @@ const FlowPanel: FC<Props> = ({id, controls, persistentControls, children}) => {
...
                     color={
                       id === focused
                         ? ComponentColor.Secondary
                         : ComponentColor.Default
                     }
-                    titleText="Configure"
                     className="flows-config-panel-button"
-                  />
+                    testID="square-button"
+                  ></Button>
```
Comment (Asalem1) [inline src/flows/components/panel/FlowPanel.tsx:163]: why like this? Why not have the button be a self-closing tsx?

File: src/flows/components/Sidebar.tsx (hunk 1 above)
Comment (ShmuelLotman) [reply src/flows/components/Sidebar.tsx:58]: Not needed

File: src/flows/components/Sidebar.tsx (hunk 2 above)
Comment (ShmuelLotman) [reply src/flows/components/Sidebar.tsx:225]: Absolute is necessary for it to be positioned the way it is relative to the panel (pun intended). This isn't really related to clockface positioning or styling, so i didn't use clockface variables, but i can move this into a css file, for sure

File: src/flows/components/panel/FlowPanel.tsx (hunk 4 above)
Comment (ShmuelLotman) [reply src/flows/components/panel/FlowPanel.tsx:163]: Good question lol, ill change that

File: src/flows/components/Sidebar.tsx (hunk 2 above)
Comment (Asalem1) [reply src/flows/components/Sidebar.tsx:225]: gotcha.

**Review by Asalem1 — APPROVED** [review]

**Pull request merged**